File: CSharpCodeBase/components/character/enemysupportcomponent.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local EnemySupportComponent = Class({__includes=Component, init=function(self, entity)
      Component.init(self, entity)
end})

function EnemySupportComponent:OnFriendAttacked(data)
  self.entity:WakeUp()
  if not self.entity.relationship:HasInstance("enemy", data.target) then
    if math.chance(self.entity.config:Get('supportScreamChance')) then
      self.entity._screamTrigger = true
      self.entity._screamTarget = data.target
    end
  end
  self.entity.relationship:AddInstance("enemy", data.target)
end

return EnemySupportComponent

File: CSharpCodeBase/components/character/enemysupportcomponent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using UnityEngine;
using System.Collections;
using MainGame.core;

namespace MainGame {
    public class EnemySupportComponent : MonoBehaviour {

        //FIXME
        public void OnFriendAttacked(object[] data) {
            //this.entity:WakeUp();
            /*if(!this.entity.relationship:HasInstance("enemy", data.target)  ){ 
              if(math.chance(this.entity.config:Get("supportScreamChance"))  ){ 
                this.entity._screamTrigger = true;
                this.entity._screamTarget = data.target;
              }
            }
            this.entity.relationship:AddInstance("enemy", data.target);*/
        }
    }
}